code/_onclick/click.dm

```diff
@@ -141,6 +144,22 @@
 		else
 			RangedAttack(A,params)
 
+//Is the atom obscured by a PREVENT_CLICK_UNDER object above it
+/atom/proc/IsObscured()
+	if(!isturf(loc)) //This only makes sense for things directly on turfs for now
+		return FALSE
+	var/turf/T = loc
+	for(var/atom/movable/AM in T)
+		if(AM.flags & PREVENT_CLICK_UNDER && AM.density && AM.layer > layer)

```

Comment:
I thought this looked interesting enough to port, but noted that there should probably be a ```plane``` check here as well.